src/main/scala/go/format/FEN.scala

```diff
@@ -0,0 +1,71 @@
+package strategygames.go.format
+
+import strategygames.Player
+import strategygames.go.variant.Variant
+
+final case class FEN(value: String) extends AnyVal {
+
+  override def toString = value
+
+  def board: String = removePockets(value.split(' ').take(1).mkString(""))
+
+  def player: Option[Player] =
+    value.split(' ').lift(1) flatMap (_.headOption) flatMap Player.apply
+
+  def player1Score: Int = intFromFen(3).getOrElse(0)
+
+  def player2Score: Int = intFromFen(4).getOrElse(0)
+
+  def fullMove: Option[Int] = intFromFen(6)
+
+  def ply: Option[Int] =
+    fullMove map { fm =>
+      fm * 2 - (if (player.exists(_.p1)) 2 else 1)
+    }
+
+  def komi: Double = intFromFen(5).getOrElse(0) / 10.0
+
+  def handicap: Option[Int] = if (fullMove == Some(1)) Some(board.count(_ == 'S')) else None
+
+  def engineFen: String =
+    removePockets(value.split(' ').take(3).mkString(" "))
+      .replace("S", "X")
+      .replace("s", "O")
+      .replace("19", "199") // goengine cant handle double digits
+      .replace("18", "189") // todo does this mess up ko?
```

Comment:
just to check youve thought about this?

Reply:
I'm not sure the ko rule is working correctly anyways, this is something I would like Paul to test, and/or provide him some examples on dev for the cases I've looked at. 
this string replace could mess up the ko string as it uses a coordinate, e.g. a18 would become a189 which is wrong. However so far the engine has always returned '-' so need to investigate this really.....

Reply:
re made to avoid ko changing